htsget-search/src/storage/local.rs

```diff
@@ -139,7 +140,70 @@ mod tests {
     });
   }
 
-  // TODO add tests for `LocalStorage::url`
+  #[test]
+  fn url_of_non_existing_key() {
+    with_local_storage(|storage| {
+      let result = storage.url("non-existing-key", UrlOptions::default());
+      assert_eq!(
+        result,
+        Err(StorageError::InvalidKey("non-existing-key".to_string()))
+      );
+    });
+  }
+
+  #[test]
+  fn url_of_folder() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder", UrlOptions::default());
+      assert_eq!(result, Err(StorageError::NotFound("folder".to_string())));
+    });
+  }
+
+  #[test]
+  fn url_with_forbidden_path() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder/../../passwords", UrlOptions::default());
+      assert_eq!(
+        result,
+        Err(StorageError::InvalidKey(
+          "folder/../../passwords".to_string()
+        ))
+      );
+    });
+  }
+
+  #[test]
+  fn url_of_existing_key() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder/../key1", UrlOptions::default());
```

Comment:
Why is this allowed though? @chris-zen ?

Reply:
As I understood it is allowed to use `..` as long as the resulting path still starts with  `storage.base_path()`. I don't know if it is really useful.

Reply:
Exactly, we don't allow to escape from `storage.base_path()`, what users do within is their problem.